### ISSUE TITLE
Clj-kondo warns on valid multi-arity defn with 2nd attr-map

### ISSUE THREAD

**Issue by lread** [clj-kondo/clj-kondo#1433]
**version**
v2021.09.25

.**platform**
native

**problem**
The multi-arity syntaxes of defn and defmacro accept an optional 2nd metadata attribute map.
When the 2nd attribute map is present, clj-kondo issues a warning when it should not.

**repro**
```
❯ clj-kondo --config '{:config-paths ^:replace []}' --lint - <<< '(defn x ([]) ([x] x) {:added "attr-map2"})'
<stdin>:1:22: error: Function arguments should be wrapped in vector.
linting took 8ms, errors: 1, warnings: 0
```

**config**
none

**expected behavior**
Clj-kondo should not warn here.

**next steps**
I shall follow up with a PR
